File: Forge1.20.1/src/main/java/online/afeibaili/mchat/MessageListener.java
package online.afeibaili.mchat;

import cn.afeibaili.mchat.message.MessageType;
import net.minecraftforge.event.ServerChatEvent;
import net.minecraftforge.event.entity.player.PlayerEvent;
import net.minecraftforge.eventbus.api.SubscribeEvent;

import java.util.Optional;

import static online.afeibaili.mchat.MChat.config;

/**
 * # 消息监听类
 *
 * @author AfeiBaili
 * @version 2026/7/19 22:00
 */

public class MessageListener {
    @SubscribeEvent
    public void onMessage(ServerChatEvent event) {
        String source = "[" + config.getName() + "] " + event.getUsername();
        Optional.ofNullable(MChat.client).ifPresent(client -> client.send(new MessageType.Text(source, event.getMessage().getString(), "")));
    }

    @SubscribeEvent
    public void onPlayerLoginIn(PlayerEvent.PlayerLoggedInEvent event) {
        Optional.ofNullable(MChat.client).ifPresent(client -> client.send(new MessageType.Text(event.getEntity().getName().getString(), "加入了" + config.getName(), "")));
    }

    @SubscribeEvent
    public void onPlayerLoginOut(PlayerEvent.PlayerLoggedOutEvent event) {
        Optional.ofNullable(MChat.client).ifPresent(client -> client.send(new MessageType.Text(event.getEntity().getName().getString(), "退出了" + config.getName(), "")));
    }
}

File: NeoForge1.21.1/src/main/kotlin/online/afeibaili/mchat/MessageListener.kt
package online.afeibaili.mchat

import cn.afeibaili.mchat.message.MessageType
import net.neoforged.bus.api.SubscribeEvent
import net.neoforged.neoforge.event.ServerChatEvent
import net.neoforged.neoforge.event.entity.player.PlayerEvent
import online.afeibaili.mchat.MChat.config


/**
 * # 消息监听器
 *
 * @author AfeiBaili
 * @version 2026/7/18 20:25
 */

class MessageListener {
    @SubscribeEvent
    fun onMessage(event: ServerChatEvent) {
        runCatching {
            MChat.client?.send(
                MessageType.Text(
                    "[${config!!.name}] ${event.username}",
                    event.message.string,
                    config!!.channel
                )
            )
        }
    }

    @SubscribeEvent
    fun onPlayerLoginIn(event: PlayerEvent.PlayerLoggedInEvent) {
        runCatching {
            MChat.client?.send(
                MessageType.Text(
                    "${event.entity.name.string}",
                    "加入了${config!!.name}",
                    config!!.channel
                )
            )
        }
    }

    @SubscribeEvent
    fun onPlayerLoginOut(event: PlayerEvent.PlayerLoggedOutEvent) {
        runCatching {
            MChat.client?.send(
                MessageType.Text(
                    "${event.entity.name.string}",
                    "退出了${config!!.name}",
                    config!!.channel
                )
            )
        }
    }
}